src/core/thread/mlr_manager.cpp

```diff
@@ -67,10 +68,36 @@ MlrManager::MlrManager(Instance &aInstance)
 void MlrManager::HandleNotifierEvents(Events aEvents)
 {
 #if OPENTHREAD_CONFIG_MLR_ENABLE
+    uint8_t                aSequenceNumber = 0;
+    otBackboneRouterConfig config;
+    if (Get<BackboneRouter::Leader>().GetConfig(config))
+    {
+        aSequenceNumber = config.mSequenceNumber;
+    }
+
     if (aEvents.Contains(kEventIp6MulticastSubscribed))
     {
+        mRecordedSequenceNumber = aSequenceNumber;
         UpdateLocalSubscriptions();
     }
+
+    if (aEvents.Contains(kEventThreadNetdataChanged))
+    {
+        if (mRecordedSequenceNumber != aSequenceNumber)
+        {
+            mRecordedSequenceNumber = aSequenceNumber;
+            // Find all Registered Multicast Addresses, and set the state of them to `kMlrStateToRegister`
+            for (Ip6::Netif::ExternalMulticastAddress &addr :
+                 Get<ThreadNetif>().IterateExternalMulticastAddresses(Ip6::Address::kTypeMulticastLargerThanRealmLocal))
+            {
+                if (addr.GetMlrState() == kMlrStateRegistered)
+                {
+                    addr.SetMlrState(kMlrStateToRegister);
+                }
+            }
+            UpdateLocalSubscriptions();
```

Comment:
`UpdateProxiedSubscriptions` shall be also called if `OPENTHREAD_CONFIG_TMF_PROXY_MLR_ENABLE` is enabled.